tests/testthat/test-sprintf_linter.R

```diff
@@ -0,0 +1,62 @@
+context("sprintf_linter")
+test_that("returns the correct linting", {
+  expect_lint("sprintf('hello')",
+    NULL,
+    sprintf_linter)
+
+  expect_lint("sprintf('hello', 1)",
+    NULL,
+    sprintf_linter)
+
+  expect_lint("sprintf('hello %d', 1)",
+    NULL,
+    sprintf_linter)
+
+  expect_lint("sprintf('hello %d', x)",
+    NULL,
+    sprintf_linter)
+
+  expect_lint("sprintf('hello %d', x + 1)",
+    NULL,
+    sprintf_linter)
+
+  expect_lint("sprintf('hello %d', f(x))",
+    NULL,
+    sprintf_linter)
+
+  expect_lint("sprintf('hello %d', 'a')",
+    list(message = rex("invalid format '%d'; use format %s for character objects")),
+    sprintf_linter)
+
+  expect_lint("sprintf('hello %d', 1.5)",
+    list(message = rex("invalid format '%d'; use format %f, %e, %g or %a for numeric objects")),
+    sprintf_linter)
+
+  expect_lint("sprintf('hello %d',)",
+    list(message = rex("argument is missing, with no default")),
```

Comment:
Can we provide a more tailored error message?
Something like "format specification needs X more arguments" springs to mind.

Reply:
A more tailored error message is certainly better. In fact, all the lint messages come from `sprintf()` errors such as

```r
> sprintf("%d", "a")                                                                                                                                                                                                          
Error in sprintf("%d", "a") : 
  invalid format '%d'; use format %s for character objects

> sprintf("%d", 1.5)                                                                                                                                                                                                          
Error in sprintf("%d", 1.5) : 
  invalid format '%d'; use format %f, %e, %g or %a for numeric objects
```

When the arguments are numerous, it is still ambiguous which causes the error since the error messages do not say anything about which argument is wrong.

To make the errors more informative, we might need a parser of the format string to understand what is expected as inputs.